tasks/broad/ImputationTasks.wdl

```diff
@@ -0,0 +1,752 @@
+    String output_vcf_basename
+    String gatk_docker
+  }
+
+  Int disk_size = ceil(3*size(input_vcfs, "GiB"))
+
+  command <<<
+    gatk GatherVcfs \
+    -I ~{sep=' -I ' input_vcfs} \
+    -O ~{output_vcf_basename}.vcf.gz
+
+    if [ ! -f ~{output_vcf_basename}.vcf.gz.tbi ]; then
+      gatk IndexFeatureFile -I ~{output_vcf_basename}.vcf.gz
+    fi
+  >>>
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "16 GiB"
+  }
+  output {
+    File output_vcf = "~{output_vcf_basename}.vcf.gz"
+    File output_vcf_index = "~{output_vcf_basename}.vcf.gz.tbi"
+  }
+}
+
+task UpdateHeader {
+  input {
+    File vcf
+    File vcf_index
+    File ref_dict
+    String basename
+    Int disk_size = ceil(4*(size(vcf, "GiB") + size(vcf_index, "GiB"))) + 20
+    String gatk_docker
+  }
+  command <<<
+
+    ## update the header of the merged vcf
+    gatk UpdateVCFSequenceDictionary \
+    --source-dictionary ~{ref_dict} \
+    --output ~{basename}.vcf.gz \
+    --replace -V ~{vcf} \
+    --disable-sequence-dictionary-validation
+  >>>
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "8 GiB"
+  }
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+    File output_vcf_index = "~{basename}.vcf.gz.tbi"
+  }
+}
+
+task RemoveSymbolicAlleles {
+  input {
+    File original_vcf
+    File original_vcf_index
+    String output_basename
+    Int disk_size = ceil(3*(size(original_vcf, "GiB") + size(original_vcf_index, "GiB")))
+    String gatk_docker
+  }
+  command {
+    gatk SelectVariants -V ~{original_vcf} -xl-select-type SYMBOLIC -O ~{output_basename}.vcf.gz
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GiB"
+  }
+}
+
+task SeparateMultiallelics {
+  input {
+    File original_vcf
+    File original_vcf_index
+    String output_basename
+    Int disk_size =  ceil(2*(size(original_vcf, "GiB") + size(original_vcf_index, "GiB")))
+    String bcftools_docker
+  }
+  command {
+    bcftools norm -m - ~{original_vcf} -Oz -o ~{output_basename}.vcf.gz
+    bcftools index -t ~{output_basename}.vcf.gz
+  }
+  output {
+    File output_vcf = "~{output_basename}.vcf.gz"
+    File output_vcf_index = "~{output_basename}.vcf.gz.tbi"
+  }
+  runtime {
+    docker: bcftools_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GiB"
+  }
+}
+
+task OptionalQCSites {
+  input {
+    File input_vcf
+    File input_vcf_index
+    String output_vcf_basename
+    String bcftools_vcftools_docker
+  }
+  Int disk_size = ceil(2*(size(input_vcf, "GiB") + size(input_vcf_index, "GiB")))
+
+  command <<<
+    # site missing rate < 5% ; hwe p > 1e-6
+    vcftools --gzvcf ~{input_vcf}  --max-missing 0.05 --hwe 0.000001 --recode -c | bgzip -c > ~{output_vcf_basename}.vcf.gz
```

Comment:
Would we ever want to adjust `--max-missing` or `--hwe` ?

Reply:
seems possible, though I don't expect we will.  But I will make them input variables.